runelite-client/src/main/java/net/runelite/client/plugins/boosts/BoostsPlugin.java

```diff
@@ -282,18 +284,20 @@ private void updateShownSkills()
 				shownSkills.addAll(BOOSTABLE_NON_COMBAT_SKILLS);
 				break;
 		}
+		updateBoostedStats();
 	}
 
 	private void updateBoostedStats()
 	{
 		// Reset is boosted
 		isChangedDown = false;
 		isChangedUp = false;
+		skillsToDisplay.clear();
 
 		// Check if we are still boosted
 		for (final Skill skill : Skill.values())
 		{
-			if (!BOOSTABLE_COMBAT_SKILLS.contains(skill) && !BOOSTABLE_NON_COMBAT_SKILLS.contains(skill))
+			if (!shownSkills.contains(skill))
```

Comment:
Thinking about this more, it would be nice to have `isChangedUp` and `isChangedDown` set regardless of whether the given skill should be shown, and gate the +/- restore timer based on whether any skills will be shown (and the "ALWAYS" timer config). That way, if you changed your config at some point after the timer is known, you wouldn't have to wait another full cycle for another buff/debuff for the plugin to figure it out again. I'll look at this later today, I think it should be pretty straightforward to include that behavior.

Reply:
Actually, that problem is a fair bit harder to solve, and is a use case which I don't think will be encountered often enough to be worth implementing.